File: job_finder_app/job_finder_app/lib/Model/Employer.dart
import 'package:job_finder_app/Model/Specialization.dart';
import 'package:job_finder_app/Model/User.dart';

class Employer {
  int? id;
  int? userId;
  String? companyName;
  String? companyPhone;
  String? companyAddress;
  int? specializationId;
  String? companyLogo;
  var isVerified;
  User? user;
  Specialization? specialization;

  Employer({
    this.id,
    this.userId,
    this.companyName,
    this.companyPhone,
    this.companyAddress,
    this.specializationId,
    this.companyLogo,
    this.isVerified,
    this.user,
    this.specialization,
  });

  Employer.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    userId = json['user_id'];
    companyName = json['company_name'];
    companyPhone = json['company_phone'];
    companyAddress = json['company_address'];
    specializationId = json['specialization_id'];
    companyLogo = json['company_logo'];
    isVerified = json['is_verified'];
    user = json['user'] != null ? new User.fromJson(json['user']) : null;
    specialization =
        json['specialization'] != null
            ? new Specialization.fromJson(json['specialization'])
            : null;
  }
}

File: JobFinder/app/Models/Employer.php
<?php
namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Employer extends Model
{
    use HasFactory;
    public $timestamps = false;
    protected $hidden = [
        'created_at',
        'updated_at',
    ];
    protected $fillable = [
        'user_id',
        'company_name',
        'company_phone',
        'company_address',
        'specialization_id',
        'company_logo',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function specialization()
    {
        return $this->belongsTo(Specialization::class);
    }

}
